torch/csrc/autograd/functions/basic_ops.cpp

```diff
@@ -16,11 +16,10 @@ auto Error::apply(variable_list&& inputs) -> variable_list {
 }
 
 auto DelayedError::apply(variable_list&& inputs) -> variable_list {
-  tensor_list outputs;
+  variable_list outputs;
   outputs.reserve(inputs.size());
   for (auto& var : inputs) {
-    // FIXME: share version counters
-    outputs.emplace_back(var.defined() ? var.tensor_data() : at::Tensor());
+    outputs.emplace_back(var.defined() ? var : Variable());
```

Comment:
is this correct?  In the other call to wrap_outputs, we pass `variable_data`.  When is this called?  Did you just want to reuse the version counters?